catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/DynamicSchemaResolver.java

```diff
@@ -718,19 +714,9 @@ private void addToFieldsCache(Set<AttributeDescriptor> descriptors) {
   }
 
   private byte[] serialize(MetacardType anywhereMType) throws MetacardCreationException {
-    ByteArrayOutputStream baos = new ByteArrayOutputStream();
-
     try {
-      ObjectOutputStream out = new ObjectOutputStream(baos);
-      out.writeObject(anywhereMType);
-
-      byte[] bytes = baos.toByteArray();
-
-      baos.close();
-      out.close();
-
-      return bytes;
-    } catch (IOException e) {
+      return OBJECT_MAPPER.writeValueAsBytes(anywhereMType);
```

Comment:
❓  Where is the serializer defined? Why are we relying on a deserializer but not a serializer?

❗️ It would seem we are relying on reflection to serialize and a deserializer to deserialize. This can lead to problems as reflection would potentially serialize a lot more than what the interface defines. In addition, we are tightly coupling the field names to the Json format.

Reply:
The `DynamicSchemaResolver` does a defensive copy of the `MetacardType` to make sure we can always recreate it in case the original bundle that sent the metacard type is removed or unavailable.

https://github.com/codice/ddf/blob/fc121d0e87f7aa0245a341d52894eef49b7ff3ef/catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/DynamicSchemaResolver.java#L360-L362

I meant to make the object mapper use the getters instead of reflection.  Originally I figured we would just add a `StdSerializer` if we ever changed changed that implementation in a break way but I need tests to catch that.  Instead of adding those tests, I will add the `StdSerializer` instead.